fud2/src/main.rs

```diff
@@ -183,6 +207,51 @@ fn build_driver() -> Driver {
         |e, input, output| {
             e.build_cmd(&[output], "calyx", &[input], &[])?;
             e.arg("backend", "firrtl")?;
+            e.arg("args", "--emit-primitive-extmodules")?;
+            Ok(())
+        },
+    );
+
+    let firrtl_primitives_setup = bld.setup("FIRRTL with primitives", |e| {
+        e.rule("external-to-ref", "sed 's/@external([0-9]*)/ref/g' $in | sed 's/@external/ref/g' > $out")?;
```

Comment:
The awesome power of sed!!!!

Reply:
It's definitely up there in my list of favorite commands :smile: